### PR TITLE
[Breaking] Make Waypoint a named export.

### LINKED ISSUE #238: .default is back in 8.0
Unfortunately adding the `modules` property to package.json in #234 brought #221 back. We're using webpack 3.10.0

This is especially painful when trying to use both Node.js for server side rendering and the browser. 

I fixed this by adding `mainFields: ['browser', 'main']` to our webpack config though that's obviously not optimal.

### PR DESCRIPTION
**Pull request by realityking**
Fixes #238

### CODE REVIEW

**Comment by realityking** [comment]
I think this is necessary if you wanna ship an ESM. Otherwise users using webpack but not ESM will always need a workaround.

**Comment by lencioni** [comment]
@realityking there are some merge conflicts that need to be resolved here and I'd like to include this in the next release. Can you please rebase this for me?

Also, what do you think about adding the named export in addition to the default export, at least for one version. That would prevent this from being a breaking change and would allow for people to migrate onto the named export more easily.

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ yarn add react-waypoint
 ## Usage
 
 ```jsx
-import Waypoint from 'react-waypoint';
+import {Waypoint} from 'react-waypoint';
```
Comment (lencioni) [inline README.md]: nit: please add spaces inside these curlies

**Comment by lencioni** [comment]
Would you be down for whipping up a quick codemod that people can use to migrate to the named export?

**Comment by realityking** [comment]
Absolutely but it'd probably have to wait until the weekend. The rest of my week is looking quite busy.

**Pull request merged**